sound/soc/soc-core.c

```diff
@@ -419,7 +419,8 @@ static void soc_free_pcm_runtime(struct snd_soc_pcm_runtime *rtd)
 
 	list_del(&rtd->list);
 
-	flush_delayed_work(&rtd->delayed_work);
+	if (rtd->delayed_work.timer.function)
```

Comment:
We probably shouldnt be invading the ctx of the work struct. We should only use the API, that being said this was solved upstream by using a proxy function to init the delayed work right away that will call a function pointer when we discover which function to use.